File: extract_colors.py
from PIL import Image
from collections import Counter

def get_dominant_colors(image_path, num_colors=5):
    try:
        image = Image.open(image_path)
        image = image.convert('RGB')
        image = image.resize((150, 150))      # Resize for speed
        pixels = list(image.getdata())
        counts = Counter(pixels)
        dominant = counts.most_common(num_colors)
        
        print("Dominant Colors:")
        for color, count in dominant:
            hex_color = '#{:02x}{:02x}{:02x}'.format(*color)
            print(f"{hex_color}")
            
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    get_dominant_colors('public/logo.png')

File: extract_colors.js
import { Jimp } from 'jimp';

async function getDominantColors(imagePath) {
    try {
        const image = await Jimp.read(imagePath);
        const width = image.bitmap.width;
        const height = image.bitmap.height;
        const colorCounts = {};

        image.scan(0, 0, width, height, function (x, y, idx) {
            const r = this.bitmap.data[idx + 0];
            const g = this.bitmap.data[idx + 1];
            const b = this.bitmap.data[idx + 2];
            const hex = "#" + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);

            colorCounts[hex] = (colorCounts[hex] || 0) + 1;
        });

        // Sort by frequency
        const sortedColors = Object.entries(colorCounts)
            .sort(([, a], [, b]) => b - a)
            .slice(0, 5)
            .map(([color]) => color);

        console.log("Dominant Colors:", sortedColors);
    } catch (err) {
        console.error("Error:", err);
    }
}

getDominantColors('public/logo_original.png');
